core/lib/multivm/src/tracers/call_tracer/mod.rs

```diff
@@ -38,4 +52,25 @@ impl CallTracer {
         let cell = self.result.as_ref();
         cell.set(result).unwrap();
     }
+
+    fn push_call_and_update_stats(&mut self, call: FarcallAndNearCallCount) {
+        let near_calls_after = call.near_calls_after;
+        self.stack.push(call);
+
+        self.max_stack_depth = self
+            .max_stack_depth
+            .max(self.stack.len() + near_calls_after);
```

Comment:
It's definitely not correct, because you don't take into account that all stack calls have this near_call_after.